### DISCUSSION TITLE
Android privacy: In search of possibilities using `adb`

### DISCUSSION THREAD

**Discussion by undergroundwires** [undergroundwires/privacy.sexy#359] [Q&A]
Android support will be added (#324) without any definitive timelines when the collection reaches some maturity.

🔍🧐 Please provide any docs/projects about what we can configure regarding privacy using `adb` on Android.

**Comment by ItsIgnacioPortal** [comment]
> [!WARNING]
> I have not tested these commands.

A couple recommendations:

# [Performance] Disable transition animation scales, improving device performance by reducing animation overhead.
```
adb shell settings put global transition_animation_scale 0
```

# [Security] Disable 2G connectivity, protecting the device against [Stingrays](https://www.bleepingcomputer.com/news/security/android-users-can-now-disable-2g-to-block-stingray-attacks/) aka IMSI Catchers.
```
adb shell settings put global preferred_network_mode 9
```
The `preferred_network_mode` table ([source](https://cs.android.com/android/platform/superproject/main/+/main:packages/services/Telephony/src/com/android/phone/settings/RadioInfo.java;l=129;drc=de40a3f44e01829e821bebb49047b4332c176b9b)):
```java
private static final String[] PREFERRED_NETWORK_LABELS = {
        "GSM/WCDMA preferred",
        "GSM only",
        "WCDMA only",
        "GSM/WCDMA auto (PRL)",
        "CDMA/EvDo auto (PRL)",
        "CDMA only",
        "EvDo only",
        "CDMA/EvDo/GSM/WCDMA (PRL)",
        "CDMA + LTE/EvDo (PRL)",
        "GSM/WCDMA/LTE (PRL)",
        "LTE/CDMA/EvDo/GSM/WCDMA (PRL)",
        "LTE only",
        "LTE/WCDMA",
        "TDSCDMA only",
        "TDSCDMA/WCDMA",
        "LTE/TDSCDMA",
        "TDSCDMA/GSM",
        "LTE/TDSCDMA/GSM",
        "TDSCDMA/GSM/WCDMA",
        "LTE/TDSCDMA/WCDMA",
        "LTE/TDSCDMA/GSM/WCDMA",
        "TDSCDMA/CDMA/EvDo/GSM/WCDMA ",
        "LTE/TDSCDMA/CDMA/EvDo/GSM/WCDMA",
        "NR only",
        "NR/LTE",
        "NR/LTE/CDMA/EvDo",
        "NR/LTE/GSM/WCDMA",
        "NR/LTE/CDMA/EvDo/GSM/WCDMA",
        "NR/LTE/WCDMA",
        "NR/LTE/TDSCDMA",
        "NR/LTE/TDSCDMA/GSM",
        "NR/LTE/TDSCDMA/WCDMA",
        "NR/LTE/TDSCDMA/GSM/WCDMA",
        "NR/LTE/TDSCDMA/CDMA/EvDo/GSM/WCDMA",
        "Unknown"
};
```
By setting the preferred network mode to a value that excludes 2G (GSM) networks, you effectively disable 2G communications.
- `GSM` = 2G
- `WCDMA`, `CDMA`, `EvDo`  = 3G
- `TDSCDMA` = [Chinese 3G](https://www.rfwireless-world.com/Tutorials/TD-SCDMA-tutorial.html)
- `LTE` = 4G
- `NR` = 5G NR

> [!WARNING]
> The `PREFERRED_NETWORK_LABELS` varies across the different Android versions. Code 9 in Android 8 is not the same as code 9 in Android 14.

> [!NOTE]
> This protects the device against Stingrays, but [Hailstorms](https://www.cloudwards.net/how-to-block-stingray-surveillance/) are still effective.


# [Privacy] Enable Private DNS with AdGuard:
```
adb shell settings put global private_dns_mode hostname
adb shell settings put global private_dns_specifier dns.adguard-dns.com
```

Using private DNS allows DNS queries to be sent in a non-vulnerable-to-espionage way to the DNS resolver. Using the AdGuard servers, also enables users to block ads.

**Reply by mziehm** [reply]
I like the private DNS there are also alternative servers to adguard blocking trackers and ads e.g.
adblock.dns.mullvad.net (info: https://mullvad.net/en/help/dns-over-https-and-dns-over-tls)
dnsforge.de (info: https://dnsforge.de/)
but I think having also script option with a server/servers which don't block (maximum compatibility), but also doesn't log requests and doesn't filter them is also good: e.g.
unfiltered.adguard-dns.com (info: https://adguard-dns.io/de/public-dns.html)
dns.mullvad.net (info: https://mullvad.net/en/help/dns-over-https-and-dns-over-tls)
dns3.digitalcourage.de (info: https://digitalcourage.de/support/zensurfreier-dns-server)

All of the above is related to DNS over TLS (DoT) which often causes problems in company networks, where internal names cannot be resolved then.

with information from https://www.kuketz-blog.de/empfehlungsecke/#dns (in German)

**Reply by undergroundwires** [reply]
I'm not sure about encouraging any third-party commercial company solution.
This may kick-back if these organizations do anything harmful to privacy.

**Comment by mziehm** [comment]
Warning: I have not tested these commands.

# Android 12: Getting the old WiFi toggle back under Quick Settings

Since Android 12, the WiFi quick toggle button has been changed. A tap no longer activates or deactivates the WiFi interface; instead, a selection of different network connections is displayed. Only setting the slider changes the connectivity - but it can be adjusted via ADB:
```
adb shell 'settings put global settings_provider_model false'
adb shell 'settings put secure sysui_qs_tiles "wifi,cell,$(settings get secure sysui_qs_tiles)"'
```

Reversion code:
```
adb shell 'settings put global settings_provider_model true'
adb shell 'settings put secure sysui_qs_tiles "internet,$(settings get secure sysui_qs_tiles)"'
```
Translated with DeepL.com (free version) from https://www.kuketz-blog.de/android-12-alten-wifi-toggle-unter-quick-settings-wiederbekommen/

**Reply by undergroundwires** [reply]
This will be added. Thank you. It would be good with more than more UI related privacy enhancement script so "UI for Privacy" category that will contain this can have more scripts.

**Comment by poepoepoee** [comment]
> [!NOTE]  
> Some of these commands were tested on Samsung devices. They focus on disabling or uninstalling non-essential bloat apps via ADB for priva debloating.

### Remove Bloatware (Galaxy S10 / Android 12 style)
These packages were primarily tested on Samsung devices running Android 12.


To restore an uninstalled package, you can use this command:

``adb shell cmd package install-existing <package_name>``
(Replace <package_name> with the full package name found in the previous step, e.g., com.google.android.apps.photos).



#### General / System
- ``pm uninstall -k --user 0 com.android.bookmarkprovider``
- ``pm uninstall -k --user 0 com.android.dreams.basic``
- ``pm uninstall -k --user 0 com.android.dreams.phototable``
- ``pm uninstall -k --user 0 com.android.egg``
- ``pm uninstall -k --user 0 com.android.wallpaper.livepicker``
- ``pm uninstall -k --user 0 com.android.providers.partnerbookmarks``
- ``pm uninstall -k --user 0 com.android.vpndialogs``
- ``pm uninstall -k --user 0 com.sec.android.easyonehand``
- ``pm uninstall -k --user 0 com.sec.android.daemonapp``

#### Samsung / OEM Apps
- ``pm uninstall -k --user 0 com.sec.android.app.shealth``
- ``pm uninstall -k --user 0 com.samsung.android.arzone``
- ``pm uninstall -k --user 0 com.samsung.android.video``
- ``pm uninstall -k --user 0 com.sec.android.app.samsungapps``
- ``pm uninstall -k --user 0 com.samsung.android.dynamiclock``
- ``pm uninstall -k --user 0 com.samsung.android.tvplus``
- ``pm uninstall -k --user 0 com.samsung.android.lool``
- ``pm uninstall -k --user 0 com.samsung.android.messaging``
- ``pm uninstall -k --user 0 com.samsung.android.oneconnect``
- ``pm uninstall -k --user 0 com.samsung.android.voc``
- ``pm uninstall -k --user 0 com.samsung.android.calendar``
- ``pm uninstall -k --user 0 com.samsung.android.app.voicenote``
- ``pm uninstall -k --user 0 com.samsung.android.app.dressroom``
- ``pm uninstall -k --user 0 com.samsung.android.scloud``
- ``pm uninstall -k --user 0 com.samsung.android.sdk.handwriting``
- ``pm uninstall -k --user 0 com.samsung.android.universalswitch``
- ``pm uninstall -k --user 0 com.samsung.safetyinformation``
- ``pm uninstall -k --user 0 com.samsung.storyservice``
- ``pm uninstall -k --user 0 com.samsung.android.service.aircommand``
- ``pm uninstall -k --user 0 com.samsung.android.app.aodservice``

#### Samsung Pass / Pay
- ``pm uninstall -k --user 0 com.samsung.android.samsungpassautofill``
- ``pm uninstall -k --user 0 com.samsung.android.samsungpass``
- ``pm uninstall -k --user 0 com.samsung.android.spay``
- ``pm uninstall -k --user 0 com.samsung.android.spayfw``

#### AR / Stickers
- ``pm uninstall -k --user 0 com.samsung.android.aremoji``
- ``pm uninstall -k --user 0 com.sec.android.mimage.avatarstickers``
- ``pm uninstall -k --user 0 com.samsung.android.aremojieditor``
- ``pm uninstall -k --user 0 com.samsung.android.stickercenter``
- ``pm uninstall -k --user 0 com.samsung.android.app.camera.sticker.facearavatar.preload``

#### Facebook
- ``pm uninstall -k --user 0 com.facebook.system``
- ``pm uninstall -k --user 0 com.facebook.appmanager``
- ``pm uninstall -k --user 0 com.facebook.services``

#### Printing
- ``pm uninstall -k --user 0 com.android.bips``
- ``pm uninstall -k --user 0 com.google.android.printservice.recommendation``

#### Samsung Games
- ``pm uninstall -k --user 0 com.samsung.android.game.gametools``
- ``pm uninstall -k --user 0 com.samsung.android.game.gos``

#### Edge / DeX
- ``pm uninstall -k --user 0 com.samsung.android.service.peoplestripe``
- ``pm uninstall -k --user 0 com.samsung.android.app.appsedge``
- ``pm uninstall -k --user 0 com.sec.android.dexsystemui``
- ``pm uninstall -k --user 0 com.sec.android.desktopmode.uiservice``
- ``pm uninstall -k --user 0 com.sec.android.app.desktoplauncher``


You can also put ``disable`` instead which is a safer method


source: [https://xdaforums.com/t/galaxy-s10-s10-debloat-bloatware-removal-list.3912073/](url)
